system/CompletionEvent_tests.cpp

```diff
@@ -276,6 +279,42 @@ void try_synchronizing_spawns() {
   BOOST_CHECK_EQUAL(total, N);
 }
 
+void try_iterative_spmd_gce() {
+
+  // set up re-enrolling across iterations
+  gce.enroll(cores());
+  gce.reenroll_on_complete();
```

Comment:
That makes sense. One annoying thing about the GCE interface is that it has this dynamic "master core" idea that is not exposed to the user. I don't know what to write in the comments. Core 0? I guess I can make that more explicit.

But sure, ```enroll_recurring()``` sounds fine. Again there's a question of if it can only be called on the master core or if we should support recurring enrolls anywhere.